config/config.go

```diff
@@ -27,6 +27,13 @@ type Config struct {
 	Plugins                      map[string]json.RawMessage `json:"plugins"`
 	DefaultDecision              *string                    `json:"default_decision"`
 	DefaultAuthorizationDecision *string                    `json:"default_authorization_decision"`
+	MetricsProvider              *MetricsProviderConfig     `json:"metrics_provider"`
```

Comment:
To avoid having to refactor this later when someone inevitably want >1 metrics provider 😢 can we structure this more as a map or list?

Just brainstorming, but something more along the lines of:

```yaml
metrics:
  prometheus:
    enabled: true
    include_decision_info_or_something: true
    <whatever other config>
```
or maybe
```yaml
metrics:
  providers:
    - name: prometheus
      <whatever other config>
```
It is a bit more verbose but will help future proof the config a little bit.

Reply:
This is by design. The whole idea was that there can be only one active metrics provider that is configured through the initial OPA config and never changes. There is a function that sets (one) current provider. Provider registers `/metrics` endpoint, something another provider would also like to do for its own metrics format. Besides, I can hardly imagine any use case for having more than one. Do you see any?
Also even if one wants to use 2 tools together it would more make sense to write a plugin that combines metrics from two tools into one format rather than having 2 sets of metric in different format and probably overlapping.

Reply:
considering there can only be one provider in the config, do you see anything wrong with the proposed config structure?

Reply:
Alright, having noodled on this for a while I can't think of any good reason for anyone to have >1 😄 so lets go ahead with this as-is. 👍